Lib/test/_test_multiprocessing.py

```diff
@@ -6275,6 +6278,80 @@ def test_atexit(self):
                 self.assertEqual(f.read(), 'deadbeef')
 
 
+class _TestSpawnedSysPath(BaseTestCase):
+    """Test that sys.path is setup in forkserver and spawn processes."""
+
+    ALLOWED_TYPES = ('processes',)
+
+    def setUp(self):
+        self._orig_sys_path = list(sys.path)
+        self._temp_dir = tempfile.mkdtemp(prefix="test_sys_path-")
```

Comment:
You can use `TemporaryDirectory` or `os_helper.temp_cwd` with `enterContext`.

Reply:
similarly for this one, since i'm using setUp and tearDown I don't think the context manager helps much for this test. Good for use within single test cases though.